pennylane/tape/tapes/tape.py

```diff
@@ -857,17 +858,37 @@ def data(self):
     def data(self, params):
         self.set_parameters(params, trainable_only=False)
 
-    def copy(self):
-        """Returns a shallow copy of the quantum tape."""
-        tape = self.__class__()
-        tape._prep = self._prep.copy()
-        tape._ops = self._ops.copy()
-        tape._measurements = self._measurements.copy()
+    def copy(self, deep=False, tape_cls=None):
+        """Returns a copy of the quantum tape.
 
-        tape._update()
+        Args:
+            deep (bool): If True, a deep copy of the quantum tape is returned.
+                If not, a shallow copy is returned; both the copied tape and the
+                original tape will share the same operation instances.
+            tape_cls (.QuantumTape): Cast the copied tape to a specific quantum tape subclass.
```

Comment:
Good question. I'm not actually sure. The parent class of a class is `type`, so perhaps it makes sense to have `type` here, but this could be confusing to the developer.

Reply:
This is the closest question I could find: https://stackoverflow.com/questions/33768004/sphinx-docstring-specify-class-type-instead-of-object

It looks like the solution in this case was to also use `type`, e.g.,

```suggestion
            tape_cls (type): Cast the copied tape to a specific quantum tape subclass.
```

Reply:
Interesting. Though it seems this is technically correct, it is also completely unhelpful :laughing: